ingest/scripts/parse-fields-from-mumps-strain.py

```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env python3
+"""
+From stdin and if division or MuV_genotype or date is empty, attempt to parse them from the mumps strain name.
+
+Outputs the modified record to stdout.
+"""
+
+import argparse
+import json
+from sys import stdin, stdout, stderr
+import re
+
+# General list of country codes
+COUNTRY_CODES = [
+    'BIH', 'BRA', 'CAN', 'CHN', 'ES', 'ESP', 'ESP', 'GBR', 'IND', 'INDIA',
+    'IRQ', 'ITA', 'JPN', 'KOR', 'MEX', 'NLD', 'NLD', 'NOR', 'PAK', 'RUS',
+    'SRB', 'SWE', 'TWN', 'TWN', 'USA', 'VNM'
+]
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="If division or MuV_genotype or date is empty, attempt to parse them from the mumps strain name.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument("--strain-field", default='strain',
+        help="Strain field from which the division should be parsed.")
+    parser.add_argument("--division-field", default='division',
+        help="Division field to which the parsed division should be saved.")
+    parser.add_argument("--genotype-field", default='MuV_genotype',
+        help="MuV_genotype field to which the parsed genotype should be saved.")
+    parser.add_argument("--date-field", default='date',
+        help="Date field to which the parsed year should be saved.")
+
+    return parser.parse_args()
+
+def _parse_division_from_strain(record, strain_field):
+    strain = record.get(strain_field, '')
+    division_field = ''
+
+    # Expand pattern with list of country codes
+    pattern = rf"MuV[siS]/([^/0-9.]+)\.({'|'.join(COUNTRY_CODES)})/[0-9]{{1,2}}\.[0-9]{{2}}"
+    # Examples:
+    # Parse division 'Manitoba' from 'MuVs/Manitoba.CAN/17.17/5[G]'
+    # Parse division 'Ontario' from 'MuVi/Ontario.CAN/12.17/3[G]'
+    # Parse division 'Boras' from 'MuVs/Boras.SWE/2.20[G]'
+    # Avoid parsing: '14778' from 'MuVs/14778.SWE/0.06[G]
+    if re.match(pattern, strain):
+        match = re.match(pattern, strain)
+        division_field = match.group(1)
+
+    # Parse "NewYork" from 'MuVs/NewYork.USA/2019/38854'
+    # MuVs/NewYork.USA/2017/51025
+    if re.match(r'MuV[siS]/([^/0-9.]+)\.USA/', strain):
+        match = re.match(r'MuV[siS]/([^/0-9.]+)\.USA/', strain)
+        division_field = match.group(1)
```

Comment:
No need to call `re.match` twice.

You could use the walrus operator `:=`:

```suggestion
    if match := re.match(r'MuV[siS]/([^/0-9.]+)\.USA/', strain):
        division_field = match.group(1)
```

or my personal preference is to use `re.compile` first (not for [speed](https://stackoverflow.com/a/452143) but [readability](https://stackoverflow.com/a/453568)):

```suggestion
    pattern_usa_division = re.compile(r'MuV[siS]/([^/0-9.]+)\.USA/')
    if match := pattern.match(strain):
        division_field = match.group(1)
```

The same applies to other regular expressions in this file.